code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -188,29 +189,54 @@
 	luminosity = 7
 	layer = ABOVE_ALL_MOB_LAYER
 
+//Red/Blue Cubes
 /obj/item/device/warp_cube
 	name = "blue cube"
 	desc = "A mysterious blue cube."
 	icon = 'icons/obj/lavaland/artefacts.dmi'
 	icon_state = "blue_cube"
+	var/teleport_color = "#3FBAFD"
 	var/obj/item/device/warp_cube/linked
-
-
-//Red/Blue Cubes
+	var/teleporting = FALSE
 
 /obj/item/device/warp_cube/attack_self(mob/user)
 	if(!linked)
 		to_chat(user, "[src] fizzles uselessly.")
 		return
-	new /obj/effect/particle_effect/smoke(user.loc)
-	user.forceMove(get_turf(linked))
+	if(teleporting)
+		return
+	teleporting = TRUE
+	linked.teleporting = TRUE
+	var/turf/T = get_turf(src)
+	new /obj/effect/temp_visual/warp_cube(T, user, teleport_color, TRUE)
 	SSblackbox.add_details("warp_cube","[src.type]")
-	new /obj/effect/particle_effect/smoke(user.loc)
+	new /obj/effect/temp_visual/warp_cube(get_turf(linked), user, linked.teleport_color, FALSE)
+	var/obj/effect/warp_cube/link_holder = new /obj/effect/warp_cube(T)
+	user.forceMove(link_holder) //mess around with loc so the user can't wander around
+	sleep(2.5)
+	if(QDELETED(user))
+		qdel(link_holder)
+		return
+	if(QDELETED(linked))
+		user.forceMove(get_turf(link_holder))
+		qdel(link_holder)
+		return
+	link_holder.forceMove(get_turf(linked))
+	sleep(2.5)
```

Comment:
This is actually rather notable since you'd expect sleep to just be 'do thing at X world.time' but it's not actually that, it's something else that handles decimals(centiseconds) properly